Userland/Libraries/LibGfx/ImageFormats/JPEG2000Loader.cpp

```diff
@@ -801,6 +810,13 @@ static ErrorOr<void> decode_jpeg2000_header(JPEG2000LoadingContext& context, Rea
 
     TRY(parse_codestream_main_header(context));
 
+    auto size_from_siz = IntSize { context.siz.width, context.siz.height };
+    if (size_from_siz != context.size) {
+        // FIXME: If this is common, warn and use size from SIZ marker.
```

Comment:
I really hope that it never happens in practice. That would break ImageDecoder's invariant of always having the correct size available.

Reply:
Oh my bad, I was convinced this code was further down and thus after the header.